server/public/model/user.go

```diff
@@ -1042,3 +1042,7 @@ type UsersWithGroupsAndCount struct {
 	Users []*UserWithGroups `json:"users"`
 	Count int64             `json:"total_count"`
 }
+
+func (u *User) EmailDomain() string {
+	return strings.Split(u.Email, "@")[1]
+}
```

Comment:
Yes that can be abused because while on the client side we don't allow the `"dfdf@dasd"@gooo.com` email to be set when signing up, server side we don't restrict it. "dfdf@dasd"@gooo.com is a valid email per RFC: https://datatracker.ietf.org/doc/html/rfc2822#section-3.4.1

Suggestion: 

```suggestion
func (u *User) EmailDomain() string {
    at := strings.LastIndex(u.Email, "@")
    // at >= 0 holds true and this is not checked here. It holds true, because during signup we run `mail.ParseAddress(email)` 
    return email[at+1:]
}
```